Tracking/include/Tracking/Sim/MeasurementCalibrator.h

```diff
@@ -53,35 +54,34 @@ class LdmxMeasurementCalibrator {
   /// @tparam parameters_t Track parameters type
   /// @param gctx The geometry context (unused)
   /// @param trackState The track state to calibrate
-  void calibrate(
-      const Acts::GeometryContext& /*gctx*/,
-      Acts::MultiTrajectory<Acts::VectorMultiTrajectory>::TrackStateProxy
-          trackState) const {
-    ActsExamples::IndexSourceLink sourceLink =
-        trackState.getUncalibratedSourceLink()
-            .get<ActsExamples::IndexSourceLink>();
-
+  template <typename traj_t>
+  void calibrate(const Acts::GeometryContext& /*gctx*/,
+                 const Acts::CalibrationContext& /*cctx*/,
+                 const Acts::SourceLink& genericSourceLink /*sourceLink*/,
+                 traj_t::TrackStateProxy trackState) const {
+    ActsExamples::IndexSourceLink sourceLink{
+        genericSourceLink.get<ActsExamples::IndexSourceLink>()};
     assert(m_measurements and
            "Undefined measurement container in LdmxMeasurementCalibrator");
     assert((sourceLink.index() < m_measurements->size()) and
            "Source link index is outside the container bounds in "
            "LdmxMeasurementCalibrator");
 
     auto meas = m_measurements->at(sourceLink.index());
-
-    trackState.calibrated<2>().setZero();
-
     Acts::Vector2 local_pos{meas.getLocalPosition()[0],
                             meas.getLocalPosition()[1]};
-    trackState.calibrated<2>().head<2>() = local_pos;
-    // trackState.data().measdim = 2;
-    trackState.calibratedCovariance<2>().setZero();
-
-    Acts::SymMatrix2 local_cov;
+    auto tsIndexType = trackState.index();
+    auto tsCal{trackState.template calibrated<2>()};
+    auto tsCalCov{trackState.template calibratedCovariance<2>()};
+    tsCal.setZero();
+    tsCal.template head<2>() = local_pos;
+    Acts::SquareMatrix2 local_cov;
     local_cov.setZero();
     local_cov(0, 0) = meas.getLocalCovariance()[0];
     local_cov(1, 1) = meas.getLocalCovariance()[1];
-    trackState.calibratedCovariance<2>().block<2, 2>(0, 0) = local_cov;
+    tsCalCov.setZero();
+    tsCalCov.block(0, 0, 2, 2) =
+        local_cov;  // make tsCalCov 2x2 block the local_cov we just set
```

Comment:
```suggestion
     // make tsCalCov 2x2 block the local_cov we just set
    tsCalCov.block(0, 0, 2, 2) = local_cov; 
```
clang like to have the comment a line above